### PR TITLE
Cognito auth

### LINKED ISSUE #773: Switch to AWS Cognito auth
This issue is to track progress on switching to AWS Cognito as auth provider

### PR DESCRIPTION
**Pull request by orestesgaolin**
- closes #773 

Work in progress, still to do before merging and ready to review:

- [x] retest sign up and sign in flow
- [x] cleanup code,
- [x] remove unnecessary dependencies in auth related classes
- [x] fix storage of user data
- [x] configure build system to inject `awsconfiguration.json` to respective flavors and platforms
- [ ] deleting the user
- [x] getting user's address on login

Right now waiting for some fixes on backend

### CODE REVIEW

**Comment by orestesgaolin** [comment]
@Nash0x7E2 do we use mocked services for anything? For now I just throw unimplemented errors but if we use it for something then I'll implement those too

File: lib/backend/repositories/auth_repo.dart [hunk 1]
```diff
@@ -1,133 +1,119 @@
...
 import 'package:junto_beta_mobile/backend/services.dart';
-import 'package:junto_beta_mobile/hive_keys.dart';
+import 'package:junto_beta_mobile/backend/services/hive_service.dart';
+import 'package:junto_beta_mobile/models/auth_result.dart';
 import 'package:junto_beta_mobile/models/models.dart';
 
 class AuthRepo {
   const AuthRepo(
     this.authService,
     this.userRepo,
```
Comment (orestesgaolin) [inline lib/backend/repositories/auth_repo.dart:12]: idea: move userRepo outside of authRepo

**Comment by Nash0x7E2** [comment]
@orestesgaolin we were using them in the test but it's fine.

**Comment by orestesgaolin** [comment]
Merging now even though there may be some issues with password reset and delete user is missing

**Pull request merged**